event-dispatcher/src/main/java/io/sk8s/event/dispatcher/FunctionDeployer.java

```diff
@@ -17,80 +17,81 @@
 import io.sk8s.kubernetes.api.model.XFunction;
 
+import org.springframework.util.ObjectUtils;
+
 /**
+ * Makes sure a certain function is running on Kubernetes.
+ *
+ * <p>This implementation uses k8s deployments with a desired replicas count (possibly 0).</p>
+ *
+ * @author Eric Bottard
  * @author Mark Fisher
  */
-public class JobLauncher {
+public class FunctionDeployer {
 
 	private final KubernetesClient kubernetesClient;
 
 	private final ObjectMapper objectMapper = new ObjectMapper();
 
-	@Autowired
-	private EventDispatcherProperties properties;
-
-	public JobLauncher(KubernetesClient kubernetesClient) {
+	public FunctionDeployer(KubernetesClient kubernetesClient) {
 		this.kubernetesClient = kubernetesClient;
 	}
 
-	public void dispatch(XFunction functionResource) {
+	public void deploy(XFunction functionResource, int replicas) {
 		String functionName = functionResource.getMetadata().getName();
+		System.out.printf("Setting %d replicas for %s%n", replicas, functionName);
 		// @formatter:off
-		String job = this.kubernetesClient.extensions().jobs().inNamespace(this.properties.getNamespace()).createNew()
-			.withApiVersion("batch/v1")
-			.withNewMetadata()
-				.withName(functionName + "-" + System.currentTimeMillis())
-				.addNewOwnerReference()
-					.withApiVersion(functionResource.getApiVersion())
-					.withKind(functionResource.getKind())
-					.withName(functionName)
-					.withUid(functionResource.getMetadata().getUid())
-					.withController(true)
-				.endOwnerReference()
-			.endMetadata()
-			.withSpec(
-				new JobSpecBuilder()
-					.withNewTemplate()
-						.withNewMetadata()
-							.withLabels(Collections.singletonMap("function", functionName))
-						.endMetadata()
-						.withSpec(buildPodSpec(functionResource))
-					.endTemplate()
-				.build()
-			)
-			.done().toString();
+		this.kubernetesClient.extensions().deployments()
+				.inNamespace(functionResource.getMetadata().getNamespace())
+				.createOrReplaceWithNew()
+					.withApiVersion("extensions/v1beta1")
+					.withNewMetadata()
+						.withName(functionName)
+						.addNewOwnerReference()
+							.withApiVersion(functionResource.getApiVersion())
+							.withKind(functionResource.getKind())
+							.withName(functionResource.getMetadata().getName())
+							.withUid(functionResource.getMetadata().getUid())
+							.withController(true)
+						.endOwnerReference()
+					.endMetadata()
+					.withNewSpec()
+						.withReplicas(replicas)
+						.withNewTemplate()
+							.withNewMetadata()
+								.withName(functionName)
+								.withLabels(Collections.singletonMap("function", functionName))
+							.endMetadata()
+							.withSpec(buildPodSpec(functionResource))
+						.endTemplate()
+					.endSpec()
+				.done();
 		// @formatter:on
-		System.out.println("JOB: " + job);
 	}
 
 	private PodSpec buildPodSpec(XFunction function) {
 		PodSpecBuilder builder = new PodSpecBuilder()
-				.withRestartPolicy("OnFailure")
+				.withRestartPolicy("Always")
```

Comment:
this is the default for a `Deployment`

Reply:
Yeah, had "OnFailure" left over from rebase, got an error and changed explicitly to Always. Can remove.